src/measures.jl

```diff
@@ -0,0 +1,270 @@
+                        racial_pops::Array{RacePopulation, 1},
+                        proposal::AbstractProposal)
+    """ Returns only the change in measures from the last partition after
+        `proposal` was accepted.
+
+        For example, suppose district 4's new White population is 43 and
+            the new Sen2010_Dem population is 62, district 8's new White population
+            is 22 and new Sen2010_Dem population is 66. The Δ measures would
+            look like:
+                {
+                    "num_cut_edges" : partition.num_cut_edges,
+                    "dists"         : (5, 8),
+                    "White"         : [43, 22],
+                    "Sen2010_Dem"   : [62, 66],
+                }
+
+    """
+    Δ_measures = initialize_Δ_measures(partition, elections, racial_pops, proposal)
+    update_measures!(Δ_measures, proposal.D₁_nodes, graph, elections, racial_pops, 1)
+    update_measures!(Δ_measures, proposal.D₂_nodes, graph, elections, racial_pops, 2)
+    return Δ_measures
+end
+
+function get_measures(graph::BaseGraph,
+                      partition::Partition,
+                      elections::Array{Election, 1},
+                      racial_pops::Array{RacePopulation, 1},
+                      steps_taken::Int=1,
+                      proposal::AbstractProposal=DummyProposal("Optional argument used when you want Δ metrics."))
+    """ Return all the measures of interest of `partition` for election in `elections`
+        and RacePopulation in racial_pops.
+
+        Other Arguments:
+            steps_taken : Number of steps in the chain
+
+        Important:
+            The `proposal` is the step the chain just took. It is used to identify
+            only the districts just modified, so only the change in information
+            from those districts can be stored instead of all the measures.
+            If steps_taken == 1, then it returns the detailed measures instead
+            of just the Δ measures.
+    """
+    if steps_taken == 1
+        measures = get_detailed_measures(graph, partition, elections, racial_pops)
+    else
+        measures = get_Δ_measures(graph, partition, elections, racial_pops, proposal)
+    end
+    return measures
+end
+
+function get_measures_at_step(all_measures::Array{}, step)
+    """ Returns the detailed measures of the partition at step `step`.
+
+        Arguments:
+            all_measures : List of measures of partitions at each step of
+                           the Markov Chain
+            step         : The step of the chain at which measures are desired
+    """
+    if step == 1
+        return all_measures[1]
+    end
+
+    # we don't want to alter the data in all_measures
+    measures = deepcopy(all_measures[1])
+
+    for i in 2:step
+        curr_measures = all_measures[i]
+        (D₁, D₂) = all_measures[i]["dists"]
+
+        for key in keys(measures)
+            # TODO: this handling for non-array singular values is poor
```

Comment:
Leaving this TODO here for now, will come back to this when I (or you) implement partisan metrics and we have more of these values that are of length 1 (like efficiency gap)